frontend/src/views/DiagnosticEditor/index.vue

```diff
@@ -371,6 +429,53 @@ export default {
     extendedDiagnostic() {
       return this.diagnostic.diagnosticType === "COMPLETE"
     },
+    isCentralCanteen() {
+      return (
+        this.originalCanteen.productionType === "central_serving" || this.originalCanteen.productionType === "central"
+      )
+    },
+    centralKitchenDiagostic() {
+      if (this.diagnostic.year && this.originalCanteen?.centralKitchenDiagnostics)
+        return this.originalCanteen.centralKitchenDiagnostics.find((x) => x.year === this.diagnostic.year)
+      return null
+    },
+    showApproPanel() {
+      if (this.originalCanteen.productionType === "site_cooked_elsewhere" && this.centralKitchenDiagostic) {
+        return (
+          this.centralKitchenDiagostic.centralKitchenDiagnosticMode !== "APPRO" &&
+          this.centralKitchenDiagostic.centralKitchenDiagnosticMode !== "ALL"
+        )
+      }
+      return true
+    },
+    showNonApproPanels() {
+      if (this.originalCanteen.productionType === "site_cooked_elsewhere" && this.centralKitchenDiagostic)
+        return this.centralKitchenDiagostic.centralKitchenDiagnosticMode !== "ALL"
+      if (this.isCentralCanteen)
+        return this.diagnostic.centralKitchenDiagnosticMode && this.diagnostic.centralKitchenDiagnosticMode !== "APPRO"
+      return true
+    },
+    showExpansionPanels() {
+      return this.showApproPanel || this.showNonApproPanels
```

Comment:
Nitpicking : At a glance, this reads like it should always be true. Maybe you can assign a variable, or leave a comment, to explain when this isn't true?
Saying this, maybe the two computed properties and this one can be rewritten so that we have:
`approDataDelegated`
`nonApproDataDelegated`
`hasDataToEnter`

These aren't great names but that's the general gist. Instead of dictating how the variable should be used (to control the display of panels) it indicates a state of the diag/td, making logic like this clearer.